sdk/batch/Azure.ResourceManager.Batch/src/Generated/Models/BatchAccountCreateOrUpdateContent.Serialization.cs

```diff
@@ -67,7 +67,7 @@ void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
             }
             if (Optional.IsCollectionDefined(AllowedAuthenticationModes))
             {
-                if (AllowedAuthenticationModes != null)
+                if (AllowedAuthenticationModes != null && Optional.IsCollectionDefined(AllowedAuthenticationModes))
```

Comment:
This is the issue that I am tracking here: https://github.com/Azure/autorest.csharp/issues/3418